### PR TITLE
fix: Add GamutProvider Tests and Remove .storybook from coverage

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

* Adds Variable Global component
* Adds GamutProvider tests
* Add createEmotionCache tests
* Remove `.storybook` from coverage (components in there should be moved to separate package).

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.3-alpha.e686b6.0<br/>@codecademy/gamut-kit@0.3.40-alpha.e686b6.0<br/>@codecademy/gamut-labs@14.1.1-alpha.e686b6.0<br/>@codecademy/gamut-styles@9.5.4-alpha.e686b6.0<br/>@codecademy/gamut-tests@2.3.15-alpha.e686b6.0<br/>@codecademy/gamut@32.0.1-alpha.e686b6.0<br/>@codecademy/konamimojisplosion@0.2.15-alpha.e686b6.0<br/>@codecademy/markdown-overrides@0.5.27-alpha.e686b6.0<br/>@codecademy/styleguide@36.1.1-alpha.e686b6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60823b6d9371d14a287ce333--gamut-preview.netlify.app>https://60823b6d9371d14a287ce333--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60823b6d9371d14a287ce333>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.3-alpha.4a434f.0<br/>@codecademy/gamut-kit@0.3.40-alpha.4a434f.0<br/>@codecademy/gamut-labs@14.1.1-alpha.4a434f.0<br/>@codecademy/gamut-styles@9.5.4-alpha.4a434f.0<br/>@codecademy/gamut-tests@2.3.15-alpha.4a434f.0<br/>@codecademy/gamut@32.0.1-alpha.4a434f.0<br/>@codecademy/konamimojisplosion@0.2.15-alpha.4a434f.0<br/>@codecademy/markdown-overrides@0.5.27-alpha.4a434f.0<br/>@codecademy/styleguide@36.1.1-alpha.4a434f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608244b96f0f836443345bcc--gamut-preview.netlify.app>https://608244b96f0f836443345bcc--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608244b96f0f836443345bcc>Deploy Logs</a></p>

File: packages/gamut-styles/src/__tests__/GamutProvider-test.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import { createEmotionCache } from '../cache';
+import { GamutProvider } from '../GamutProvider';
+import { theme } from '../theme';
+
+jest.mock('../cache', () => {
+  const cacheMock = jest.fn();
+  return { createEmotionCache: cacheMock };
+});
+
+expect.extend(matchers);
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/src/__tests__/GamutProvider-test.tsx:16]: These matchers are great; we should apply them to all tests in a global setup file.

**Review by JoshuaKGoldberg — APPROVED** [review]
Sweet, and +1 to the coverage weirdness with storybook...

**Review by dreamwasp — APPROVED** [review]
🚀

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6082e38c07c18b5a01c32e1d--gamut-preview.netlify.app>https://6082e38c07c18b5a01c32e1d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6082e38c07c18b5a01c32e1d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-illustrations@0.10.4-alpha.70ef65.0<br/>@codecademy/gamut-kit@0.3.43-alpha.70ef65.0<br/>@codecademy/gamut-labs@14.2.3-alpha.70ef65.0<br/>@codecademy/gamut-styles@9.5.5-alpha.70ef65.0<br/>@codecademy/gamut-tests@2.3.16-alpha.70ef65.0<br/>@codecademy/gamut@32.1.3-alpha.70ef65.0<br/>@codecademy/konamimojisplosion@0.2.16-alpha.70ef65.0<br/>@codecademy/markdown-overrides@0.5.28-alpha.70ef65.0<br/>@codecademy/styleguide@36.2.3-alpha.70ef65.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6082ee49bb20946ae7ca3498--gamut-preview.netlify.app>https://6082ee49bb20946ae7ca3498--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6082ee49bb20946ae7ca3498>Deploy Logs</a></p>

**Pull request merged**